### ISSUE TITLE
XOSTOR

### ISSUE THREAD

**Issue by osuno1** [ronivay/XenOrchestraInstallerUpdater#240]
I'm sure I just don't understand, but the image does not have the ability to use XOSTOR?  When I select XOSTOR, it says "XOSTOR is available in XOA".  Is it only available in the web version and only limited?  Thanks for any help!

**Comment by vmpr** [comment]
XOSTOR is only available with a paid subscription and not in the free version. cheers Ringo

**Comment by ronivay** [comment]
Hey,

That's one of the turnkey features you get from the paid application,  ability to install and configure XOSTOR via XOA. Similar to the possibility of installing netdata to hosts via XOA and the VM template hub.

**Comment by osuno1** [comment]
Thank you both for the clarification and quick response.  I use your solution all the time and really appreciate all your efforts!

**Comment by neutor** [comment]
XOSTOR is now available for all users, not just paying ones. Install  scripts are provided by XCP-NG team:
https://xcp-ng.org/forum/topic/5361/xostor-hyperconvergence-preview

Could the option to install XOSTOR added to the XenOrchestraInstallerUpdater script?
CLI only usage is OK.
Not sure if XOSTOR GUI can be enabled in XO-from-sources, but it would be sooo nice :)

This is a great project, respect!

**Comment by neutor** [comment]
> so no not going to be added as part of this project

Yes, my mistake, it makes sense not including here the install scripts posted in the forum.
But, XOA (not open-source) does have the ability to install XOSTOR  with a click of a button. I found this post by CEO of XCP-NG (?):

Q:  Will XOSTOR eventually make it into the sources version? Currently it says XOSTOR is only available in XOA.
A: CEO and co-founder of Vates, creator of both Xen Orchestra projects here :)
     It is open source (the code is there), but the "easy/turnkey" part is now in XOA only (even XOA Free!). However, you can achieve the exact same result with the CLI and keeping your XO from the sources, if you don't want to use XOA Free (which is free as 0$). You can find the procedure that's not that hard to follow here: https://xcp-ng.org/forum/topic/5361/xostor-hyperconvergence-preview/

Which makes me think, if the code to install XOSTOR is public, in XOA-Free (not open-source), can your script add it to XO from-sources? 
XOSTOR will be essential part of XCP-NG experience, Shared Storage using local drives!. I just bought three Lenovo M720Qs and added 2TB SSD to each :)

Cheers!

**Comment by ronivay** [comment]
> > so no not going to be added as part of this project
> 
> Yes, my mistake, it makes sense not including here the install scripts posted in the forum.
> 
> But, XOA (not open-source) does have the ability to install XOSTOR with a click of a button. I found this post by CEO of XCP-NG (?):
> 
> _Q: Will XOSTOR eventually make it into the sources version? Currently it says XOSTOR is only available in XOA. A: CEO and co-founder of Vates, creator of both Xen Orchestra projects here :) It is open source (the code is there), but the "easy/turnkey" part is now in XOA only (even XOA Free!). However, you can achieve the exact same result with the CLI and keeping your XO from the sources, if you don't want to use XOA Free (which is free as 0$). You can find the procedure that's not that hard to follow here: https://xcp-ng.org/forum/topic/5361/xostor-hyperconvergence-preview/_
> 
> Source: https://old.reddit.com/r/xcpng/comments/1ceeff8/difference_between_xoa_free_tier_and_xo_from/?rdt=46476
> 
> Which makes me think, if the code to install XOSTOR is public, in XOA-Free (not open-source), can your script add it to XO from-sources? XOSTOR will be essential part of XCP-NG experience, Shared Storage using local drives!. I just bought three Lenovo M720Qs and added 2TB SSD to each :)
> 
> EDIT: Bought another M720Q lol
> 
> Cheers!

The ability to do so is part of the added benefit you get from a licensed XOA. Similar to the template hub and automated netdata installation to the host which aren't part of the XO from sources. If these are ever "officially" supported by the XO from sources, i've no problem to add them here but until then i have no interest to even try to hack them in especially the stuff that directly modifies the host side.

**Comment by neutor** [comment]
>   ... netdata installation to the host which aren't part of the XO from sources

Just so I understand, please. This means XO-from-sources is missing some code, available only in licensed XOA? Does it mean that some parts of XOA are closed-source? 

When he says "It is open source (the code is there), but the "easy/turnkey" part is now in XOA only", what "XOA only" means exactly, please?  We cannot have it in XO-from-sources?

I'm surely missing something...

**Comment by neutor** [comment]
I think I got it. It is possible, like removing nagging messages, but "might hurt XO and the XCP-ng project" (XO docs). Makes sense.
I'm onto hacking together my own installer now. Sorry for confusion. Thanks again for great project!
